crates/ruff_python_parser/src/parser/pattern.rs

```diff
@@ -583,22 +583,22 @@ impl<'src> Parser<'src> {
                     has_seen_pattern = false;
                     has_seen_keyword_pattern = true;
 
+                    // Even in case the key pattern is invalid, we still need to parse
+                    // the value pattern to move the parser forward.
+                    let value_pattern = parser.parse_match_pattern();
```

Comment:
What I understand from this comment is that without this `parse_match_pattern` call it's possible that the list parsing gets stuck because the `item` parsing function never progress. Is this what it means? Or does it mean that we should always try to parse the patch pattern value because we're after the `=` and we know that we should find a value pattern.

Reply:
The former. Taking the example from the PR description, the parser being at `1`, but it expects a comma thus it breaks out of parsing the list with an error.

This makes me wonder about the case where there's no pattern after `=` like in the following snippet:
```python
match foo:
	case Foo(bar=, baz):
		pass
```

It adds an error here:

https://github.com/astral-sh/ruff/blob/5e0e1f09793508eef3bfac136df42ee865b55e72/crates/ruff_python_parser/src/parser/pattern.rs#L526-L529

Not sure why it's "Expression expected.", I'll change it to "Expected a pattern" as by this time we've tried all possible pattern parsing.

Reply:
Okay, I think we have a different understanding of `stuck`. To me, stuck means that the parser goes into an infinite loop. I dont' think that's the case here because the error recovery bails out if it doesn't parse the `value_pattern`. However, it doesn't recover nicely from the error.

However, there's a fundamental problem with this parsing method now that we always call `parse_match_pattern`... we drop the `value_pattern` in case it isn't a `PatternMatchAs`. I don't think we should be doing this. Can we instead convert `pattern` into ` PatternMatchAs` if we've found a `=`?

Reply:
From Discord:

> So, the `PatternKeyword` node structure (for keyword patterns like `case Foo(a=1): ...`) is that it contains an identifier and a pattern. We can use an empty `Expr::Name` pattern here and retain the parsed value pattern. This would be similar to the empty `Identifier` patten.